collect_app/src/main/res/layout/color_picker_dialog_layout.xml

```diff
@@ -0,0 +1,359 @@
+            android:id="@+id/current_color"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:padding="@dimen/margin_small"
+            android:layout_marginTop="@dimen/margin_extra_large"
+            app:layout_constraintWidth_percent="0.2"
+            app:layout_constraintDimensionRatio="w,1:1"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/color11" >
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:background="@drawable/color_circle"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"/>
+        </androidx.constraintlayout.widget.ConstraintLayout>
+
+        <com.google.android.material.textview.MaterialTextView
+            style="@style/TextAppearance.Collect.Headline5"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/hex"
+            app:layout_constraintBottom_toBottomOf="@+id/materialTextView"
+            app:layout_constraintStart_toEndOf="@+id/current_color"
+            app:layout_constraintTop_toTopOf="@+id/materialTextView" />
+
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/materialTextView"
+            style="@style/TextAppearance.Collect.Headline5"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginHorizontal="@dimen/margin_small"
+            android:text="@string/hash"
+            app:layout_constraintBottom_toBottomOf="@+id/hex_color_layout"
+            app:layout_constraintEnd_toStartOf="@+id/hex_color_layout"
+            app:layout_constraintTop_toTopOf="@+id/hex_color_layout" />
+
+        <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/hex_color_layout"
+            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+            android:theme="@style/Theme.AndroidShared.MaterialComponents.Polyfill.TextInputLayout"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            app:hintEnabled="false"
+            app:layout_constraintTop_toTopOf="@+id/current_color"
+            app:layout_constraintBottom_toBottomOf="@+id/current_color"
+            app:layout_constraintStart_toStartOf="@+id/color14"
+            app:layout_constraintEnd_toEndOf="parent">
+
+            <com.google.android.material.textfield.TextInputEditText
+                android:id="@+id/hex_color"
+                style="@style/TextAppearance.Collect.Headline5"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:maxLength="6"/>
```

Comment:
I added this because hex colors consist of 6 chars. The thing is that users might ignore that `#` we have before this editText and try to add it. I thought that maybe we could resolve it adding a `TextWatcher` that would automatically remove it. Or we can change this value to 7 (or even remove at all) and try to use both formats (with and without #).

@lognaturel what do you thing? It's probably something for polishing after beta but I wanted to bring it up?

Reply:
I don't think we need to specifically worry about the `#`. I do think that before beta we should prevent a permanent crash when characters outside the hex range are provided (https://github.com/getodk/collect/pull/4600#issuecomment-861016973). I've put some notes for a longer-term approach at https://github.com/getodk/collect/issues/4606.